Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.inl

```diff
@@ -92,6 +94,28 @@ template<class DataTypes>
 void BilateralInteractionConstraint<DataTypes>::init()
 {
     unspecializedInit();
+
+    if (sofa::core::topology::BaseMeshTopology* _topology1 = l_topology1.get())
+    {
+        m1.createTopologyHandler(_topology1);
+        m1.addTopologyEventCallBack(core::topology::TopologyChangeType::POINTSREMOVED,
+            [this](const core::topology::TopologyChange* change)
+        {
+            const auto* pointsRemoved = static_cast<const core::topology::PointsRemoved*>(change);
+            removeContact(0, pointsRemoved->getArray());
+        });
+    }
+
+    if (sofa::core::topology::BaseMeshTopology* _topology2 = l_topology2.get())
+    {
+        m2.createTopologyHandler(_topology2);
+        m2.addTopologyEventCallBack(core::topology::TopologyChangeType::POINTSREMOVED,
+            [this](const core::topology::TopologyChange* change)
+        {
+            const auto* pointsRemoved = static_cast<const core::topology::PointsRemoved*>(change);
+            removeContact(1, pointsRemoved->getArray());
+        });
+    }
```

Comment:
yes because you could have a bilateral constraint between a dynamic topology and a static one (like a grid).
If one of the topology is supporting change, it will properly remove the cosntraint between the 2 topo.